src/main/java/com/systelab/seed/infrastructure/security/SecurityResponseHeadersFilter.java

```diff
@@ -10,139 +9,164 @@
-        addDirectiveToContentSecurityPolicy(contentSecurityPolicy, OBJECT_SRC, objectSrc);
-        addDirectiveToContentSecurityPolicy(contentSecurityPolicy, MEDIA_SRC, mediaSrc);
-        addDirectiveToContentSecurityPolicy(contentSecurityPolicy, FRAME_SRC, frameSrc);
-        addDirectiveToContentSecurityPolicy(contentSecurityPolicy, REPORT_URI, reportUri);
-        addSandoxDirectiveToContentSecurityPolicy(contentSecurityPolicy, sandboxValue);
-        response.getHeaders().putSingle(CONTENT_SECURITY_POLICY_HEADER,contentSecurityPolicy.toString());
-    }
-
-    private void addDirectiveToContentSecurityPolicy(StringBuilder contentSecurityPolicy, String directiveName, String value) {
-        if (isNotBlank(value) && !defaultSrc.equals(value)) {
-            contentSecurityPolicy.append("; ").append(directiveName).append(" ").append(value);
-        }
-    }
-
-    private void addSandoxDirectiveToContentSecurityPolicy(StringBuilder contentSecurityPolicy, String value) {
-        if (isNotBlank(value)) {
-            if ("true".equalsIgnoreCase(value)) {
-                contentSecurityPolicy.append("; ").append(SANDBOX);
-            } else {
-                contentSecurityPolicy.append("; ").append(SANDBOX).append(" ").append(value);
-            }
-        }
+  /**
+   * Enables a sandbox for the requested resource similar to the iframe sandbox attribute. The
+   * sandbox applies a same origin policy, prevents popups, plugins and script execution is blocked.
+   * You can keep the sandbox value empty to keep all restrictions in place, or add values:
+   * allow-forms allow-same-origin allow-scripts, and allow-top-navigation
+   */
+  public static final String SANDBOX = "sandbox";
+  // The default policy for loading content such as JavaScript, Images, CSS, Font's, AJAX requests, Frames, HTML5 Media
+  public static final String DEFAULT_SRC = "default-src";
+  // Defines valid sources of images
+  public static final String IMG_SRC = "img-src";
+  // Defines valid sources of JavaScript
+  public static final String SCRIPT_SRC = "script-src";
+  // Defines valid sources of stylesheets
+  public static final String STYLE_SRC = "style-src";
+  // Defines valid sources of fonts
+  public static final String FONT_SRC = "font-src";
+  // Applies to XMLHttpRequest (AJAX), WebSocket or EventSource
+  public static final String CONNECT_SRC = "connect-src";
+  // Defines valid sources of plugins, eg <object>, <embed> or <applet>.
+  public static final String OBJECT_SRC = "object-src";
+  // Defines valid sources of audio and video, eg HTML5 <audio>, <video> elements
+  public static final String MEDIA_SRC = "media-src";
+  // Defines valid sources for loading frames
+  public static final String FRAME_SRC = "frame-src";
+  public static final String REPORT_URI = "report-uri";
+
+  public static final String SELF_REFERENCE = "'self'";
+  public static final String ACCESS_CONTROL_ALLOW_ORIGIN = "Access-Control-Allow-Origin";
```

Comment:
Why are all that fields public? Could they be private?

Reply:
You are right. It was copied "as is" from the previous class. I have set all of them private and moved the values of the directives inside the method.